src/test/java/controllers/FloorControllerTests.java

```diff
@@ -0,0 +1,138 @@
+    public void configureSliderDeviceTest() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        Device device = new Device();
+        device.setTopicName("topic1");
+        JFXSlider slider = new JFXSlider();
+        slider.setUserData(device);
+        Method initDevicesMethod = FloorController.class.getDeclaredMethod("configureSliderDevice", JFXSlider.class);
+        initDevicesMethod.setAccessible(true);
+        Capture<Topic> deviceTopic = newCapture();
+        //Record
+        model.subscribe(capture(deviceTopic));
+        replay(model);
+        //Play
+        initDevicesMethod.invoke(controller, slider);
+        //Verify
+        verify(model);
+        assertEquals(device.getTopicName(), deviceTopic.getValue().getTopicName());
+    }
+
+    @Test
+    public void configureToggleDeviceTest() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        Device device = new Device();
+        device.setTopicName("topic1");
+        JFXToggleButton toggleButton = new JFXToggleButton();
+        toggleButton.setUserData(device);
+        Method initDevicesMethod = FloorController.class.getDeclaredMethod("configureToggleDevice", JFXToggleButton.class);
+        initDevicesMethod.setAccessible(true);
+        Capture<Topic> deviceTopic = newCapture();
+        //Record
+        model.subscribe(capture(deviceTopic));
+        replay(model);
+        //Play
+        initDevicesMethod.invoke(controller, toggleButton);
+        //Verify
+        verify(model);
+        assertEquals(device.getTopicName(), deviceTopic.getValue().getTopicName());
+    }
+
+    @Test
+    public void configureBrightnessDeviceTest() throws NoSuchFieldException, IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+        Topic deviceTopic = new Topic("brightnessTopic");
+        Double initBrightnessSlider = 50.0;
+        Double finalBrightnessSlider = 100.0;
+        Double initBrightness = 0.0;
+        Double finalBrightness = 0.5;
+
+        //Floor plan field
+        Field floorPlanField = FloorController.class.getDeclaredField("floorPlan");
+        ImageView floorPlan = new ImageView();
+        floorPlanField.setAccessible(true);
+        floorPlanField.set(controller, floorPlan);
+
+        //InitBrightnewssDevice Method
+        Method initBrightnessDevice = FloorController.class.getDeclaredMethod("initBrightnessDevice", Topic.class);
+        initBrightnessDevice.setAccessible(true);
+
+        //Test
+        deviceTopic.setLastValue(initBrightnessSlider);
+        initBrightnessDevice.invoke(controller, deviceTopic);
+        ColorAdjust effect = (ColorAdjust) floorPlan.effectProperty().get();
+        assertEquals(initBrightness, effect.getBrightness());
+        deviceTopic.setLastValue(finalBrightnessSlider);
+        assertEquals(finalBrightness, effect.getBrightness());
+    }
+
+    @Test
+    public void initControllerTest() throws IOException {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/POPBL6/front-end/file/2720643970/issues/source?bid=3316499&fileBranchId=3347636#l129)